wandb/internal/tb_watcher.py

```diff
@@ -0,0 +1,279 @@
+"""
+tensor b watcher.
+"""
+
+import json
+import os
+import threading
+import time
+
+import six
+from six.moves import queue
+import wandb
+from wandb import util
+from wandb.util import json_dumps_safer_history
+
+
+# Give some time for tensorboard data to be flushed
+SHUTDOWN_DELAY = 5
+
+
+def _link_and_save_file(path, base_path=None, sender=None):
+    # TODO(jhr): should this logic be merged with Run.save()
+    if base_path is None:
+        base_path = os.path.dirname(path)
+    files_dir = sender._settings.files_dir
+    file_name = os.path.relpath(path, base_path)
+    abs_path = os.path.abspath(path)
+    wandb_path = os.path.join(files_dir, file_name)
+    util.mkdir_exists_ok(os.path.dirname(wandb_path))
+    # We overwrite existing symlinks because namespaces can change in Tensorboard
+    if os.path.islink(wandb_path) and abs_path != os.readlink(wandb_path):
+        os.remove(wandb_path)
+        os.symlink(abs_path, wandb_path)
+    elif not os.path.exists(wandb_path):
+        os.symlink(abs_path, wandb_path)
+    # TODO(jhr): need to figure out policy, live/throttled?
+    sender._save_file(file_name)
```

Comment:
Looks familiar ;)  We should set `policy="live"` here.  It will upload every N seconds based on it's size.

Reply:
Added it, i was worried because cli-og has a throttled policy.  I left the note to look into policy improvements